src/parser.rs

```diff
@@ -1012,25 +1008,23 @@ impl<'a> Scanner<'a> {
 
             // TODO record name usage
             match seg.statements[index as usize].stype {
-                OpenGroup => {
-                    top_group = index;
-                }
+                OpenGroup => top_group = index,
                 CloseGroup => {
                     if top_group == NO_STATEMENT {
-                        self.diagnostics.push((index, Diagnostic::UnmatchedCloseGroup));
+                        self.diag(Diagnostic::UnmatchedCloseGroup);
```

Comment:
Pretty sure I was getting borrow checker errors last time I tried that, but it clearly compiles now, so let's take it